File: Programming-languages-and-laboratory/Python/Exercises/29b_compito_java_20190411/Impiegato.py
class Impiegato:
    def __init__(self, codice, tipo, nome, costo):
        self._codice = codice
        self._tipo = tipo
        self._nome = nome
        self._costo = costo
        self._numero_servizi_svolti = 0

    def incrementa_numero_servizi_svolti(self):
        self._numero_servizi_svolti += 1

    def get_codice(self):
        return self._codice

    def get_numero_servizi(self):
        return self._numero_servizi_svolti

    def get_costo(self):
        return self._costo

    def get_nome(self):
        return self._nome

    def get_costo_orario(self):
        return self._costo

    def __str__(self):
        return self._nome + "\t" + str(self._codice)

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/29_compito_java_2019_04_11b/Impiegato.java
public abstract class Impiegato {

	protected int codice;
	protected String nome;
	protected double costo;
	protected String tipo;

	public Impiegato(int codice, String nome, String tipo, double costo) {
		this.codice = codice;
		this.nome = nome;
		this.costo = costo;
		this.tipo = tipo;
	}

	public double getCosto() {
		return costo;
	}

	public int getCod() {
		return codice;
	}

	public String getNome() {
		return nome;
	}

	public String toString() {
		return nome + "\t" + codice + "\t" + tipo;
	}
}